### ISSUE TITLE
Drawing classifier evaluation object has additional keys that it should not have

### ISSUE THREAD

**Issue by srikris** [apple/turicreate#1974]
There seems to be more in this dictionary that is needed. 

```
In [23]: evals.keys()
Out[23]:
['f1_score',
 'roc_curve',
 'labels',
 'precision',
 'label_metrics',
 'sorted_labels',
 'training_time',
 'conf_mat',
 'num_test_examples',
 'confidence_threshold',
 'num_classes',
 'max_iterations',
 'confidence_metric_for_threshold',
 'training_loss',
 'auc',
 'confusion_matrix',
 'feature',
 'test_data',
 'num_examples',
 'recall',
 'hesitant_threshold',
 'model_name',
 'accuracy']
```

**Comment by znation** [comment]
@srikris Which ones should it have, and which ones should it not have?

**Comment by srikris** [comment]
I think it should only have these:

```
['f1_score',
 'roc_curve',
 'precision',
 'auc',
 'confusion_matrix',
 'recall',
 'accuracy']
```

**Comment by TobyRoseman** [comment]
This is related to #1785. Some of those additional keys/values are very costly to generate, particularly when there are a large number of classes.

**Comment by nickjong** [comment]
I think the main issue here is that many of these keys are not actually metrics, whereas previously the caller could iterate over the keys of the evaluation dictionary and assume that each key was an evaluation metric

**Comment by syoutsey** [comment]
To verify after implementing in C++.

**Comment by srikris** [comment]
Verified fix for 6.0

```
In [16]: metrics.keys()
Out[16]:
['f1_score',
 'auc',
 'recall',
 'precision',
 'log_loss',
 'roc_curve',
 'confusion_matrix',
 'accuracy']
```
